src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseWriter.cs

```diff
@@ -66,6 +66,14 @@ public override void Flush()
             _writer.Flush();
         }
 
+        public override Task FlushAsync()
+        {
+            if (IsClosed)
+                ThrowClosed();
```

Comment:
It seems that none of XmlBaseWriter's methods checked async operation in flight because the type didn't have any async methods. Does this mean I should add this check in every public sync/async method?